adapters/sspbc/sspbc.go

```diff
@@ -0,0 +1,407 @@
+package sspbc
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/golang/glog"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const version = "5.6"
+
+// MC payload (for banner ads)
+type McAd struct {
+	Id      string             `json:"id"`
+	Seat    string             `json:"seat"`
+	SeatBid []openrtb2.SeatBid `json:"seatbid"`
+}
+
+// Adslot data (oneCode detection)
+type AdSlotData struct {
+	PbSlot string `json:"pbslot"`
+	PbSize string `json:"pbsize"`
+}
+
+// Banner Template payload
+type TemplatePayload struct {
+	SiteId  string `json:"siteid"`
+	SlotId  string `json:"slotid"`
+	AdLabel string `json:"adlabel"`
+	PubId   string `json:"pubid"`
+	Page    string `json:"page"`
+	Referer string `json:"referer"`
+	McAd    string `json:"mcad"`
+}
+
+// Ext data in request.imp
+type SsbcRequestImpExt struct {
+	Data AdSlotData `json:"data"`
+}
+
+// Ext data added by proxy
+type SsbcResponseExt struct {
+	AdLabel     string `json:"adlabel"`
+	PublisherId string `json:"pubid"`
+	SiteId      string `json:"siteid"`
+	SlotId      string `json:"slotid"`
+}
+
+type SspbcAdapter struct {
```

Comment:
Please rename to simply `adapter`. There is no need to repeat the package and no need to export the type by starting with an uppercase letter. Access modifiers in Go are determined by if the first letter of the name if upper case (exported) or lower case (package private). Please change the other types to also start with a lower case letter.